### PR TITLE
Test open ssl version

### PR DESCRIPTION
**Pull request by eranha**
### Desired Outcome

CLI is shipped with openssl version > 1.1.1k.

### Implemented Changes

Upgraded ubuntu image version in docker files

### Connected Issue/Story

CyberArk internal issue link: [#14467](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-14467)

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # ========== BUILD CONTAINER ===========
-FROM ubuntu:18.04 as conjur-cli-builder
+FROM ubuntu:21.10 as conjur-cli-builder
```
Comment (mbenita-Cyberark) [inline Dockerfile:2]: just one question. did you manage to compare the build time between this version and the old one (before upgrating the ubuntu version?)

File: test/test_integration_openssl.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                               ['policy', 'replace', '-b', 'root', '-f',
+                                self.environment.path_provider.get_policy_path("initial")])
+
+    # *************** TESTS ***************
+
+    @integration_test()
+    def test_open_ssl_version_is_sufficient(self):
+        import ssl
+        version = ssl.OPENSSL_VERSION.split(' ')[1]
+        result = version in ['1.1.1k', '1.1.1l']
```
Comment (mbenita-Cyberark) [inline test/test_integration_openssl.py:34]: In second thaught lets make sure that it is a 1.1.1 and that the last letter is bigger than k

**Review by mbenita-Cyberark — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (eranha) [reply Dockerfile:2]: No

**Review by mbenita-Cyberark — APPROVED** [review]
lgtm

**Pull request merged**